store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaOverlapTest.java

```diff
@@ -154,14 +160,19 @@ public void shouldOverwriteVisibilityPropertyWhenConflictingSchemasMerge() throw
         assertThat(schema.validate().isValid())
                 .withFailMessage(schema.validate().getErrorString())
                 .isTrue();
-        assertThat(schema.getVisibilityProperty()).isEqualTo(PROPERTY_1);
+        assertThat(schema.getVisibilityProperty())
+                .withFailMessage("After a recovery from a collision with Visibility Property, the property should be set to Null")
+                .isNull();
         assertThat(schema.toString())
-                .isNotEqualTo(GRAPH_A_SCHEMA.toString())
-                .isEqualTo(graphBSchema.toString());
+                .isEqualTo(new Schema.Builder(GRAPH_A_SCHEMA)
```

Comment:
Use the built in to string comparison for `assertThat`
```java
assertThat(schema).hasToString(...);
```

Reply:
ooooh nice! 👌